iothub/device/src/Transport/HttpClientHelper.cs

```diff
@@ -532,6 +530,14 @@ public void Dispose()
                     _httpClientObj = null;
                 }
 
+                // HttpClientHandler that is used to create HttpClient will automatically be disposed when HttpClient is disposed
+                // But in case the client handler didn't end up being used by the HttpClient, we explicitly dispose it here.
+                if (_httpClientHandler != null)
+                {
+                    _httpClientHandler?.Dispose();
```

Comment:
nit: question mark here is redundant since we already check if it's not null

Reply:
I will change it in an upcoming PR since I have the gates passed already and this is safe.